app/server/api-generator/src/main/java/io/syndesis/server/api/generator/openapi/OpenApiValidationRules.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Copyright (C) 2016 Red Hat, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.syndesis.server.api.generator.openapi;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.function.Function;
+
+import io.syndesis.server.api.generator.APIValidationContext;
+import io.syndesis.server.api.generator.openapi.v2.Oas20ValidationRules;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * This class contains Syndesis custom validation rules for Open API specifications.
+ */
+public final class OpenApiValidationRules implements Function<OpenApiModelInfo, OpenApiModelInfo> {
+
+    private static final Logger LOG = LoggerFactory.getLogger(OpenApiValidationRules.class);
+
+    private final List<Function<OpenApiModelInfo, OpenApiModelInfo>> v2Rules = new ArrayList<>();
+    private final List<Function<OpenApiModelInfo, OpenApiModelInfo>> v3Rules = new ArrayList<>();
```

Comment:
I was hoping that using the Apicurio OpenAPI would not necessitate two different sets of validation rules, one for each version. I guess that's not so?

Reply:
I prepared for having different sets because I think some rules may be obsolete according to the changed spec architecture in 3.x. On the other hand we might want to add some 3.x specific rules. When I get to implementing the rules for 3.x I probably can say more about it